recipes/cc65/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+        datadir = self._datadir
+        prefix = self.package_folder
+        samplesdir = self._samplesdir
+        if tools.os_info.is_windows:
+            datadir = tools.unix_path(datadir)
+            prefix = tools.unix_path(prefix)
+            samplesdir = tools.unix_path(samplesdir)
+        return [
+            "PREFIX={}".format(prefix),
+            "datadir={}".format(datadir),
+            "samplesdir={}".format(samplesdir),
+        ]
+
+    @contextmanager
+    def _alias_x86_64_arch(self):
+        # MSBuild uses only self.settings.arch to detect the current architecture.
+        # So change the arch to x86 when it is x86_64 since these are compatible.
+        settings_modified = self.settings.copy()
+        if self.settings.arch_build == "x86_64":
+            self.output.info("x86_64 arch detected. Aliasing to x86.")
+            settings_modified.arch_build = "x86"
+            settings_modified.arch = "x86"
+        settings_original = self.settings
+        self.settings = settings_modified
+        yield
+        self.settings = settings_original
+
+    def _build_msvc(self):
+        with self._alias_x86_64_arch():
+            msbuild = MSBuild(self)
+            msvc_platforms = {
+                "x86": "Win32",
+            }
+            msbuild.build(os.path.join(self._source_subfolder, "src", "cc65.sln"),
+                          build_type="Release", platforms=msvc_platforms)  # , arch=self.settings.arch_build)
+            with tools.chdir(os.path.join(self._source_subfolder, "libsrc")):
+                self.run("{}".format(os.environ["CONAN_MAKE_PROGRAM"]))
+
+    def _build_autotools(self):
+        autotools = self._configure_autotools()
+        with tools.chdir(os.path.join(self._source_subfolder)):
+            autotools.make(args=self._make_args)
+
+    def build(self):
+        if self.settings.compiler == "Visual Studio":
+            self._build_msvc()
+        else:
+            self._build_autotools()
+
+    def _package_msvc(self):
+        self.copy("*.exe", src=os.path.join(self._source_subfolder, "bin"), dst=os.path.join(self.package_folder, "bin"), keep_path=False)
+        for dir in ("asminc", "cfg", "include", "lib", "target"):
+            self.copy("*", src=os.path.join(self._source_subfolder, dir), dst=os.path.join(self._datadir, dir))
+
+    def _package_autotools(self):
+        autotools = self._configure_autotools()
+        with tools.chdir(os.path.join(self.build_folder, self._source_subfolder)):
+            autotools.install(args=self._make_args)
+
+        tools.rmdir(self._samplesdir)
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        if self.settings.compiler == "Visual Studio":
+            self._package_msvc()
+        else:
+            self._package_autotools()
+
+    def package_id(self):
+        del self.info.settings.compiler
```

Comment:
```suggestion
        self.info.include_build_settings()
        del self.info.settings.compiler
        del self.info.settings.arch
```

Remove arch since it's not a library package

Reply:
If I don't use `arch` and run `conan create . cc65/2.18@ -s arch_build=x86`,
the output contains:
```
Cross-build from 'Windows:x86' to 'Windows:x86_64'
```
and
```
> set "VSCMD_START_DIR=%CD%" && call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC/Auxiliary/Build/vcvarsall.bat" x86_amd64 && devenv "source_subfolder\src\cc65.sln" /upgrade && msbuild "source_subfolder\src\cc65.sln" /p:Configuration="Release" /p:UseEnv=true /p:Platform="Win32" /m:8 /p:PlatformToolset="v141" /verbosity:minimal /p:ForceImportBeforeCppTargets="C:\Users\maarten\.conan\data\cc65\2.18\_\_\build\b2fcbc60b3bee75023a4f7934cadb723eec550bd\conan_build.props"
-----------------
```

Which indicates it loads the vcvars for x86_amd64.

`MSBuild.build` does always use `settings.arch=x86_64`, disregarding any argument to that function.

Reply:
@uilianries I've opened an issue at conan-io/conan#6403 because `MSBuild.build` should not choose `x86_64` by default.

Reply:
Probably it will continue as is, because MSBuild is a stable feature, and we can't change it behavior.

Reply:
@madebr I think you facing the same problem that I saw some time ago: https://github.com/conan-io/conan/issues/5004

Reply:
@uilianries Check my latest patch. I mock a settings object with the arch and os attribute copied from arch_build and os_build.
The only thing I am unhappy about this, is the usage of the private `_data` member.

Reply:
And Still no builds by CI.

Reply:
The no builds were caused by an invalid ´config.yml´. It appears the CI does not fail when this is incorrect.